### PR TITLE
[RDY] Remove char_u (6)

### PR DESCRIPTION
**Pull request by ghost**
Sorry about the size of this, but `smsg`, among other functions, was quite pervasive.

I'll try to keep these PRs smaller if it's wanted, although there's a ton of char_u usage so big batches might be a better idea.

### CODE REVIEW

File: src/nvim/ex_eval.c [hunk 1]
```diff
@@ -702,9 +701,9 @@ static void report_pending(int action, int pending, void *value)
   default:
     if (pending & CSTP_THROW) {
       vim_snprintf((char *)IObuff, IOSIZE,
-          (char *)mesg, _("Exception"));
-      mesg = vim_strnsave(IObuff, STRLEN(IObuff) + 4);
-      STRCAT(mesg, ": %s");
+                   mesg, _("Exception"));
+      mesg = (char *)vim_strnsave(IObuff, STRLEN(IObuff) + 4);
+      strcat(mesg, ": %s");
```
Comment (splinterofchaos) [inline src/nvim/ex_eval.c]: This could be written `mesg = concat_str(IObuff, ": %s");`.

File: src/nvim/main.c [hunk 2]
```diff
@@ -1174,10 +1173,10 @@ static void command_line_scan(mparm_T *parmp)
               p = xmalloc(STRLEN(a) + 4);
               sprintf((char *)p, "so %s", a);
               parmp->cmds_tofree[parmp->n_commands] = TRUE;
-              parmp->commands[parmp->n_commands++] = p;
-            } else
-              parmp->commands[parmp->n_commands++] =
-                (char_u *)argv[0];
+              parmp->commands[parmp->n_commands++] = (char *)p;
```
Comment (splinterofchaos) [inline src/nvim/main.c]: Lools like `p` is defined just a few lines above. Can it be localized and declared as a `char *`?

**Comment by splinterofchaos** [comment]
This actually seems to remove more casts than it adds. Good work!

File: src/nvim/ex_eval.c (hunk 1 above)
Comment (Pyrohh) [inline src/nvim/ex_eval.c]: :+1:

File: src/nvim/main.c (hunk 2 above)
Comment (Pyrohh) [inline src/nvim/main.c]: Yes, I declared it as `s` as I didn't want to shadow the top level one.

**Comment by Pyrohh** [comment]
> This actually seems to remove more casts than it adds. Good work!

Thanks, it's very satisfying to remove all these casts :-)
Anyways, I pushed a review commit.

**Comment by splinterofchaos** [comment]
LGTM :+1:

**Comment by Pyrohh** [comment]
@elmart Would you like to take a look? I assume you're just busy, but I'd rather be sure of that then have you come back in a few days with a bunch of issues for an already merged PR :-)

**Comment by elmart** [comment]
Sorry, guys, I've landed a new job and you what happens... 
I'll try to have a look on this tomorrow. 
If I can't, feel free to merge.

**Comment by Pyrohh** [comment]
> Sorry, guys, I've landed a new job and you what happens... 
> I'll try to have a look on this tomorrow.

Thanks for the heads up, but please only do so if you want to and have the time; I don't want anyone feeling some sort of obligation to reviewing PRs if they're busy or don't want to.

**Comment by Pyrohh** [comment]
Thanks for reviewing @splinterofchaos.

**Pull request merged**